rasa/nlu/training_data/formats/rasa_yaml.py

```diff
@@ -0,0 +1,189 @@
+import logging
+from typing import Text, Any, List, Dict, Tuple
+
+import typing
+
+from rasa.nlu.training_data.entities_parser import EntitiesParser
+from rasa.nlu.training_data.formats.readerwriter import TrainingDataReader
+import rasa.utils.io
+from rasa.nlu.training_data.lookup_tables_parser import LookupTablesParser
+from rasa.nlu.training_data.synonyms_parser import SynonymsParser
+
+if typing.TYPE_CHECKING:
+    from rasa.nlu.training_data import TrainingData
+
+
+logger = logging.getLogger(__name__)
+
+KEY_NLU = "nlu"
+KEY_INTENT = "intent"
+KEY_INTENT_EXAMPLES = "examples"
+KEY_INTENT_TEXT = "text"
+KEY_SYNONYM = "synonym"
+KEY_SYNONYM_EXAMPLES = "examples"
+KEY_REGEX = "regex"
+KEY_REGEX_EXAMPLES = "examples"
+KEY_LOOKUP = "lookup"
+KEY_LOOKUP_EXAMPLES = "examples"
+
+DOCS_LINK = "https://rasa.com/docs/rasa/"
+
+
+class RasaYAMLReader(TrainingDataReader):
+    def __init__(self) -> None:
+        self.training_examples = []
+        self.entity_synonyms = {}
+        self.regex_features = []
+        self.lookup_tables = []
+
+    def reads(self, s: Text, **kwargs: Any) -> "TrainingData":
+        from rasa.nlu.training_data import TrainingData
+
+        self.__init__()
+
+        yaml_content = rasa.utils.io.read_yaml(s)
+
+        for key, value in yaml_content.items():
+            if key == KEY_NLU:
+                self._parse_nlu(value)
+            else:
+                logger.warning(
+                    f"Unexpected key '{key}' found in {self.filename}"
+                    f"Acceptable keys are: {KEY_NLU}"
+                )
+
+        return TrainingData(
+            self.training_examples,
+            self.entity_synonyms,
+            self.regex_features,
+            self.lookup_tables,
+        )
+
+    def _parse_nlu(self, nlu_data: List[Dict[Text, Any]]) -> None:
+
+        for nlu_item in nlu_data:
+            if not isinstance(nlu_item, dict):
+                logger.warning(
+                    f"Unexpected block found in '{self.filename}': \n"
+                    f">> {nlu_item}\n"
+                    f"Items under the `nlu` key must be YAML dictionaries."
+                    f"This block will be skipped."
+                    f"Please check the docs at {DOCS_LINK}."
+                )
+                continue
+
+            if KEY_INTENT in nlu_item.keys():
+                self._parse_intent(nlu_item)
+            elif KEY_SYNONYM in nlu_item.keys():
+                self._parse_synonym(nlu_item)
+            elif KEY_REGEX in nlu_item.keys():
+                self._parse_regex(nlu_item)
+            elif KEY_LOOKUP in nlu_item.keys():
+                self._parse_lookup(nlu_item)
```

Comment:
A final `else` clause would be super helpful here, in case the user has a typo in their key (e.g. `regexp` instead of `regex`) we can show them a warning.

Reply:
In that case maybe we can grab the "first" non-`examples` non-`metadata` key in the dict and include it in the warning, explaining that it couldn't be recognized.